tortoise/models.py

```diff
@@ -1022,12 +1022,12 @@ async def get_or_create(
         if not defaults:
             defaults = {}
         db = using_db or cls._choose_db(True)
-        async with in_transaction(connection_name=db.connection_name):
+        async with in_transaction(connection_name=db.connection_name) as conn:
             instance = await cls.filter(**kwargs).first()
             if instance:
                 return instance, False
             try:
-                return await cls.create(**defaults, **kwargs, using_db=using_db), True
+                return await cls.create(**defaults, **kwargs, using_db=conn), True
```

Comment:
Just update to ` return await cls.create(**defaults, **kwargs, using_db=db), True`